In [61]:
import os
import pandas as pd

# Function to get words from a file and return as a set
def get_words(filename):
    with open(filename, 'r') as file:
        words = file.readlines()
    return set(word.strip() for word in words)

# Define the path to the folder where the files are located
folder_path = 'C:/Users/jmenu/OneDrive - MUNI/Life/MUNI/Mgr/python data/solo/dictionary' 

# List of all .txt files and unix-words to include
txt_files = ['enable1.txt', 'ospd.txt', 'popular.txt', 'unix-words']

# Initialize a set to hold all unique words
all_words = set()

# Loop through all text files and collect words
for txt_file in txt_files:
    file_path = os.path.join(folder_path, txt_file)
    words = get_words(file_path)
    all_words.update(words)  # Add words to the set to keep them unique

# Convert the set of words to a DataFrame and save it to CSV
df = pd.DataFrame(sorted(all_words), columns=['Word'])
df.set_index('Word', inplace=True)

# Save to 'words.csv'
df.to_csv('words.csv')

print("words.csv has been created with all unique words.")

words.csv has been created with all unique words.


In [147]:
import pandas as pd

# For a simple .txt file, you can use read_csv with no delimiter (if each line is a word)
df = pd.read_csv(r'C:\Users\jmenu\OneDrive - MUNI\Life\MUNI\Mgr\python data\solo\dictionary\enable1.txt', header=None, names=['Word'])

In [ ]:
#Add Columns Count and Value, also convert Word to indexes
# Function to calculate the value based on the word
def calculate_value(word):
    if isinstance(word, str):  # Ensure the word is a string
        return sum(ord(char.lower()) - ord('a') + 1 for char in word if char.isalpha())  # Process alphabetic characters
    return 0  # If it's not a valid word, return 0

# Apply the function to the 'Word' column directly, not the index

df['Value'] = df['Word'].apply(calculate_value)
df['Char Count'] = df['Word'].str.len()
df.set_index('Word', inplace=True)


In [125]:
df.head()

Word  Value  Char Count
0      aa      2         2.0
1     aah     10         3.0
2   aahed     19         5.0
3  aahing     40         6.0
4    aahs     29         4.0

How many elements does this dataframe have?

In [91]:
df.describe()
df.shape[0]

172823

What is the value of the word microspectrophotometries?

In [159]:
df.loc['pinfish', 'Value']

81.0

What is the highest possible value of a word?

In [160]:
df.describe()
df['Value'].max()

319

Which of the following words have a Char Count of 15?
nfold


glowing


pinfish


microbrew


superheterodyne

In [161]:
df.index = df.index.astype(str)
df.loc[['nfold','glowing','pinfish','microbrew','superheterodyne']]

KeyError: "['nfold'] not in index"

What is the highest possible length of a word?

What is the word with the value of 319?

What is the most common value?

What is the shortest word with value 274?

Create a column Ratio which represents the 'Value Ratio' of a word

What is the maximum value of Ratio?

What word is the one with the highest Ratio?

How many words have a Ratio of 10?

What is the maximum Value of all the words with a Ratio of 10?

Of those words with a Value of 260, what is the lowest Char Count found?¶

Based on the previous task, what word is it?